### ISSUE TITLE
Carthage Support 🙁

### ISSUE THREAD

**Issue by beamPacer** [Giphy/giphy-ios-sdk-ui-example#120]
Hi! I see that as of version 2.0.3, there's no Carthage support for this distribution. That's inconvenient for us, because we don't use Cocoapods anymore. :(

We'll fork this repo or otherwise make this work to add Carthage support back, but it's unfortunate that we have to spend the time doing this. Two questions: 1) Do you have any plans to add Carthage support back here, and 2) Is there a reason you removed it? Do you predict that our efforts to build your dependencies and make this work for our non-Cocoapods system will fail for some known reason?

Thank you!

**Comment by cgmaier** [comment]
hey @beamPacer , v sorry to hear about this. 

the vast majority of Github issues and requests from our partners over the past 6 months have related exclusively to CocoaPods integrations, and so we felt it was worthwhile to focus our efforts elsewhere instead of maintaining and testing Carthage for every release. 

unfortunately it might be a good bit of work for us to get up and running with Carthage. I'm going to take a look at what's necessary and get back to you soon.

**Comment by cgmaier** [comment]
took a look and made some progress, though it seems ive run into this issue https://github.com/Carthage/Carthage/issues/3019

specifically for compiling the dependencies `DeepDiff` and `PINCache`

**Comment by cgmaier** [comment]
hey @beamPacer I was able to support Carthage for the latest release.

1. Add the following entries to your Cartfile like so:
 ```
binary "https://s3.amazonaws.com/sdk.mobile.giphy.com/SDK/GiphySDK.json"
github "onmyway133/DeepDiff"
github "pinterest/PINCache"
 ```
 2. Add `-ObjC` to the `Other Linker Flags` Build Setting

 3. Add `PINCache.framework`, `PINOperation.framework`, `DeepDiff.framework`, `GiphyCoreSDK.framework`, `GiphyUISDK.framework` from `Carthage/Build/iOS` into your project. 

 4. Create a new Run Script under Build Phases (See step 8 of the directions [here](https://github.com/Carthage/Carthage#quick-start)). Only 2 of the frameworks need to be added to the run script (The rest can be excluded from this step as they are static frameworks):   
 - `PINCache.framework`  
 - `DeepDiff.framework` 

5. Download WebP.framework (attached below - couldn't figure out how to import via Carthage) and add to your project. 

you will have to choose a work around for this issue - there are a few different approaches as I understand it ! 
https://github.com/Carthage/Carthage/issues/3019

let me know if you have issues and I can share a sample project. 

[WebP.framework.zip](https://github.com/Giphy/giphy-ios-sdk-ui-example/files/5566938/WebP.framework.zip)

**Comment by beamPacer** [comment]
Oh that's amazing, thank you so much!! I'm not able to download the GiphyCore and GiphyUI frameworks, though -- when i add `binary "https://s3.amazonaws.com/sdk.mobile.giphy.com/SDK/GiphySDK.json"` to my cartfile and run `carthage update GiphySDK`, nothing gets added to the /Build/iOS folder. I've tried it with and without `--platform iOS` -- is this working for you, am I doing something wrong?

**Comment by beamPacer** [comment]
Oh also, looking at https://s3.amazonaws.com/sdk.mobile.giphy.com/SDK/GiphySDK.json, it looks like the last version in there is 2.0.3. Did you mean to send me a different json link?

**Comment by cgmaier** [comment]
hmm just tried again from scratch and things appear to work ok for me: 

<img width="1239" alt="Screen Shot 2020-11-19 at 12 23 17 PM" src="https://user-images.githubusercontent.com/11489602/99713970-389e6b80-2a62-11eb-8d9c-3133e4786966.png">
 
here's the sample project: 
https://www.dropbox.com/s/0wmajvmt1wg7sqp/test-carthage.zip?dl=0

**Comment by beamPacer** [comment]
No, unfortunately it's still not pulling down for me :( I wonder if it's a binary that only you have access to, something Carthage is able to pull down for you because your machine is credentialed differently than mine? Only reason I can think. Anyway, I used the version _from_ the sample project you linked, and that worked!

**Comment by cgmaier** [comment]
can you try in a blank Xcode project? don't think it's a credentials issue. I wonder if the binary download doesn't play nice if the project has a Carthage folder already all setup.

**Comment by beamPacer** [comment]
Tried it in a brand new project _and_ in your project, separately -- neither one of them download it :(

**Comment by cgmaier** [comment]
we still aren't able to reproduce this on our end - we also tested on a machine with no aws credentials 
any ideas?

**Comment by cgmaier** [comment]
hey @beamPacer it seems like there are still some major blocking issues with Carthage. they don't currently support pre-compiled .xcframeworks. it was necessary to switch to xcframework to support the new M1 Macs, as well as Mac Catalyst and SPM (upcoming)

**Comment by msilva** [comment]
Hi @cgmaier - Can you re-evaluate this based on the latest release of Carthage supporting xcframeworks?

**Comment by cgmaier** [comment]
hey @msilva my understanding is that the latest Carthage release provides compiling of xcframeworks via the `--use-frameworks` flag, but doesn't allow distributing precompiled xcframeworks. 

https://github.com/Carthage/Carthage/issues/3103

we will continue to include links to binaries in our release notes, so you may integrate them directly

**Comment by ezzz** [comment]
Hello @cgmaier 

I was using Giphy with SDK last year. Maybe you recall some of my questions :) 
I understand it is no more possible.
What should I do exactly then?
- Remove Giphy json from Carthage
- Add the folder GiphyUISDK.xcframework from your repo to my own project?
- What about PINCache and PINOperation? I leave them in Carthage?
- I saw some webp dependency. Is there something to do with it?

Currently we would like to release a new version of our iOS app but we are currently stuck with this Giphy issue.

Thanks for you support

Best regards

**Comment by cgmaier** [comment]
hey @ezzz somehow this slipped by me! very sorry for the late reply. 

a while ago I updated our docs with some info around Carthage: 
https://github.com/Giphy/giphy-ios-sdk/blob/main/Docs.md#carthage

we no longer use PINCache or PINOperation as we built our own internal caching solution.
